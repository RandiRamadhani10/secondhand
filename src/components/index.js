import BaseInput from './BaseInput';
import BaseButton from './BaseButton';
import BottomTabNavigation from './BottomTabNavigation';
import TabItem from './TabItem';
import Gap from './Gap';

export {Gap, BaseInput, BaseButton, BottomTabNavigation, TabItem};
