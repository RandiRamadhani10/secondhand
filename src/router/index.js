import React from 'react';

import {NavigationContainer} from '@react-navigation/native';

import AppRoute from './AppRoute';

const Router = () => {
  return (
    <NavigationContainer>
      <AppRoute />
    </NavigationContainer>
  );
};

export default Router;
