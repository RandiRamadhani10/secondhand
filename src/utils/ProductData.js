export const productsData = [
  {
    id: 1,
    name: 'sepatu',
    description: 'Description Sepatu',
    base_price: 1000000,
    image_url: 'https://firebasestorage.googleapis.com/v0/b/market-final-project.appspot.com/o/products%2FPR-1654962957757-sepatu.jpg?alt=media',
    image_name: 'PR-1654962957757-sepatu.jpg',
    location: 'Bandung',
    user_id: 1,
    created_at: '2000-01-01T00:00:00.000Z',
    updated_at: '2000-01-01T00:00:00.000Z'
  },
  {
    id: 2,
    name: 'baju',
    description: 'Description Baju',
    base_price: 500000,
    image_url: 'https://firebasestorage.googleapis.com/v0/b/market-final-project.appspot.com/o/products%2FPR-1654962957757-baju.jpg?alt=media',
    image_name: 'PR-1654962957757-baju.jpg',
    location: 'Bandung',
    user_id: 1,
    created_at: '2000-01-01T00:00:00.000Z',
    updated_at: '2000-01-01T00:00:00.000Z',
    categories: [
    {
        id: 1,
        name: 'Fashion'
    },
    {
        id: 2,
        name: 'Otomotif'
    },
    {
        id: 3,
        name: 'Electronic'
    },
    {
        id: 4,
        name: 'Food'
    }
    ]
  },
];
