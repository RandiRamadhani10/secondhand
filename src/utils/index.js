export * from './Colors';
export * from './Fonts';
export * from './showMessages';
export * from './ErrorCode';
export * from './Navigation';
export * from './FormatDate';
