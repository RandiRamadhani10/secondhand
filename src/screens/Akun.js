import React, {useEffect} from 'react';
import {StyleSheet, Text, SafeAreaView, View, TouchableOpacity} from 'react-native';
import {Colors, Fonts} from '../utils';
import {moderateScale} from 'react-native-size-matters';
import {Divider, Gap} from '../components';
import CameraButton from '../components/CameraButton';
import {ICEdit, ICLogout, ICSettings} from '../assets';

import {useIsFocused} from '@react-navigation/native';
import {useDispatch, useSelector} from 'react-redux';
import {clearUser} from '../store/usersSlice';

const Akun = ({navigation}) => {
  const dispatch = useDispatch();

  const isFocused = useIsFocused();
  const usersState = useSelector(state => state.users.users);

  useEffect(() => {
    if (!usersState.hasOwnProperty('access_token')) {
      navigation.navigate('Login');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isFocused]);

  return (
    <SafeAreaView style={styles.screen}>
      <Text style={styles.header}>Akun Saya</Text>
      <Gap height={moderateScale(24)} />
      <View style={styles.cameraContainer}>
        <CameraButton onPress={() => {}} />
      </View>
      <Gap height={moderateScale(24)} />

      <TouchableOpacity
        style={styles.menuAkunItem}
        activeOpacity={0.7}
        onPress={() => {
          navigation.navigate('Profile');
        }}>
        <ICEdit />
        <Gap width={moderateScale(16)} />
        <Text style={styles.menuAkunItemTitle}>Ubah Akun</Text>
      </TouchableOpacity>
      <Gap height={moderateScale(16)} />
      <Divider />

      <Gap height={moderateScale(16)} />
      <TouchableOpacity style={styles.menuAkunItem} activeOpacity={0.7} onPress={() => {}}>
        <ICSettings />
        <Gap width={moderateScale(16)} />
        <Text style={styles.menuAkunItemTitle}>Pengaturan Akun</Text>
      </TouchableOpacity>
      <Gap height={moderateScale(16)} />
      <Divider />

      <Gap height={moderateScale(16)} />
      <TouchableOpacity
        style={styles.menuAkunItem}
        activeOpacity={0.7}
        onPress={() => {
          dispatch(clearUser());
          navigation.navigate('Main', {screen: 'Home'});
        }}>
        <ICLogout />
        <Gap width={moderateScale(16)} />
        <Text style={styles.menuAkunItemTitle}>Keluar</Text>
      </TouchableOpacity>
      <Gap height={moderateScale(16)} />
      <Divider />

      <Gap height={moderateScale(16)} />
      <Text style={styles.version}>Version 1.0.0</Text>
    </SafeAreaView>
  );
};

export default Akun;

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: Colors.WHITE,
    padding: moderateScale(20),
  },
  header: {
    fontFamily: Fonts.PRIMARY.BOLD,
    fontSize: moderateScale(20),
    color: Colors.TEXT,
  },
  cameraContainer: {
    alignItems: 'center',
  },
  menuAkunItem: {
    flexDirection: 'row',
  },
  menuAkunItemTitle: {
    fontFamily: Fonts.PRIMARY.MEDIUM,
    fontSize: moderateScale(14),
    color: Colors.TEXT,
  },
  version: {
    fontFamily: Fonts.PRIMARY.REGULAR,
    fontSize: moderateScale(12),
    color: Colors.SECONDARY,
    textAlign: 'center',
  },
});
