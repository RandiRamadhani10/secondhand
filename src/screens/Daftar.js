import React from 'react';
import {StyleSheet, Text, View, SafeAreaView, StatusBar, TouchableOpacity} from 'react-native';
import {Colors, Fonts} from '../utils';

import {moderateScale} from 'react-native-size-matters';
import {ICArrowLeft} from '../assets';
import {BaseButton, BaseInput, Gap} from '../components';
import {Controller, useForm} from 'react-hook-form';

import {yupResolver} from '@hookform/resolvers/yup';

import * as yup from 'yup';

import {useDispatch, useSelector} from 'react-redux';
import {authRegister} from '../store/actions/users';

const Daftar = ({navigation}) => {
  const dispatch = useDispatch();
  const isLoading = useSelector(state => state.users.isLoading);
  const schema = yup
    .object()
    .shape({
      full_name: yup.string().min(5, 'Silahkan isi nama minimal 5 karakter').required('Silahkan isi nama lengkap'),
      email: yup.string().email('Silahkan isi alamat email yang valid').required('Silahkan isi alamat email'),
      password: yup.string().min(5, 'Silahkan isi password minimal 5 karakter').required(),
    })
    .required();

  const {
    control,
    handleSubmit,
    formState: {errors},
  } = useForm({
    resolver: yupResolver(schema),
    defaultValues: {
      full_name: '',
      email: '',
      password: '',
    },
  });

  const onSubmit = async data => {
    let payload = {
      ...data,
      phone_number: 0,
      address: '-',
      image: 'test.png',
      city: '-',
    };

    dispatch(authRegister(payload));
  };

  return (
    <SafeAreaView style={styles.screen}>
      <StatusBar backgroundColor={Colors.WHITE} />
      <View>
        <TouchableOpacity activeOpacity={0.7} onPress={() => navigation.navigate('Main', {screen: 'Home'})}>
          <ICArrowLeft />
        </TouchableOpacity>
        <Gap height={40} />
        <Text style={styles.header}>Daftar</Text>
        <Gap height={24} />
        <Controller
          control={control}
          rules={{required: true}}
          render={({field: {onChange, onBlur, value}}) => (
            <BaseInput
              label="Nama"
              type="text"
              placeholder="Nama Lengkap"
              onChangeText={onChange}
              onBlur={onBlur}
              value={value}
            />
          )}
          name="full_name"
        />
        {errors?.full_name && <Text style={styles.errors}>{errors.full_name.message}</Text>}
        <Gap height={16} />
        <Controller
          control={control}
          rules={{required: true}}
          render={({field: {onChange, onBlur, value}}) => (
            <BaseInput
              label="Email"
              type="text"
              placeholder="Contoh: johndee@gmail.com"
              onChangeText={onChange}
              onBlur={onBlur}
              value={value}
            />
          )}
          name="email"
        />
        {errors?.email && <Text style={styles.errors}>{errors.email.message}</Text>}
        <Gap height={16} />
        <Controller
          control={control}
          rules={{required: true}}
          render={({field: {onChange, onBlur, value}}) => (
            <BaseInput
              label="Buat Password"
              type="password"
              placeholder="Buat Password"
              onChangeText={onChange}
              onBlur={onBlur}
              value={value}
            />
          )}
          name="password"
        />
        {errors?.password && <Text style={styles.errors}>{errors.password.message}</Text>}
        <Gap height={16} />
        <BaseButton disable={isLoading} isLoading={isLoading} title="Daftar" onPress={handleSubmit(onSubmit)} />
      </View>
      <View style={styles.loginContainer}>
        <View style={styles.loginWrapper}>
          <Text style={styles.loginSub}>Sudah punya Akun? </Text>
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={() => {
              navigation.navigate('Login');
            }}>
            <Text style={styles.loginCTA}>Masuk disini</Text>
          </TouchableOpacity>
        </View>
      </View>
    </SafeAreaView>
  );
};

export default Daftar;

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: Colors.BACKGROUND,
    padding: moderateScale(20),
  },
  header: {
    fontFamily: Fonts.PRIMARY.BOLD,
    fontSize: moderateScale(24),
    color: Colors.TEXT,
  },
  loginContainer: {
    flex: moderateScale(1),
    justifyContent: 'flex-end',
  },
  loginWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  loginSub: {
    fontFamily: Fonts.PRIMARY.REGULAR,
    fontSize: moderateScale(14),
    color: Colors.TEXT,
  },
  loginCTA: {
    fontFamily: Fonts.PRIMARY.BOLD,
    fontSize: moderateScale(14),
    color: Colors.PRIMARY,
  },
  errors: {
    fontFamily: Fonts.PRIMARY.REGULAR,
    fontSize: moderateScale(10),
    color: Colors.ERROR,
    paddingBottom: moderateScale(8),
  },
});
