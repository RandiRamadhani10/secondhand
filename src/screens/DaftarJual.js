import React, {Fragment, useEffect, useState} from 'react';
import {SafeAreaView, StyleSheet, Text, View, ScrollView, FlatList} from 'react-native';
import {
  Gap,
  CardUser,
  BaseNotif,
  CategoryButtonItem,
  ProductItem,
  ProductItemSkeleton,
  NotificationItemSkeleton,
  BaseTambahProduk,
} from '../components';
import {ICBox, ICBoxActive, ICDollarSign, ICDollarSignActive, ICLove, ICLoveActive, ILPeople} from '../assets';
import {Colors} from '../utils';
import {moderateScale} from 'react-native-size-matters';
import {Fonts} from '../utils';
import {useIsFocused} from '@react-navigation/native';
import {useDispatch, useSelector} from 'react-redux';
import {getOrder, getProduct} from '../store/actions/seller';

const DaftarJual = ({navigation}) => {
  const dispatch = useDispatch();

  const isFocused = useIsFocused();

  const [categorySelected, setCategorySelected] = useState('produk');

  const [listData, setListData] = useState([]);

  const [numColumns] = useState({
    produk: 2,
    diminati: 1,
    terjual: 1,
  });

  const usersState = useSelector(state => state.users);

  const isLoadingSellerState = useSelector(state => state.seller.isLoading);

  const getData = async tabSelected => {
    if (tabSelected === 'produk') {
      const res = await dispatch(getProduct());

      if (res) {
        setListData(res?.payload);
      }
    } else if (tabSelected === 'diminati') {
      const res = await dispatch(getOrder({status: ''}));

      if (res) {
        setListData(res?.payload);
      }
    } else if (tabSelected === 'terjual') {
      const res = await dispatch(getOrder({status: 'accepted'}));

      if (res) {
        setListData(res?.payload);
      }
    }
  };

  useEffect(() => {
    getData(categorySelected);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [categorySelected, isFocused]);

  const renderList = tabSelected => {
    if (tabSelected === 'produk') {
      return (
        <FlatList
          key={numColumns.produk}
          numColumns={numColumns.produk}
          data={[{}, ...listData]}
          showsVerticalScrollIndicator={false}
          keyExtractor={item => item.id}
          ListEmptyComponent={<BaseTambahProduk />}
          renderItem={({item, index}) => {
            return isLoadingSellerState ? (
              <View key={index} style={styles.loadingContainer}>
                <ProductItemSkeleton />
              </View>
            ) : index === 0 ? (
              <BaseTambahProduk onPress={() => navigation.navigate('Jual')} />
            ) : (
              <ProductItem
                key={index}
                title={item?.name}
                image={item?.image_url}
                category={item.Categories}
                price={item?.base_price}
                onPress={() => navigation.navigate('DetailProduk', {id: item.id})}
              />
            );
          }}
        />
      );
    } else if (tabSelected === 'diminati') {
      return (
        <FlatList
          key={numColumns.diminati}
          numColumns={numColumns.diminati}
          style={styles.listDiminati}
          data={listData}
          showsVerticalScrollIndicator={false}
          keyExtractor={item => item.id}
          ListEmptyComponent={
            <View style={styles.emptyContent}>
              <Gap height={moderateScale(50)} />
              <ILPeople />
              <Gap height={moderateScale(16)} />
              <Text style={styles.emptyText}>
                Belum ada produkmu yang diminati nih, sabar ya rejeki nggak kemana kok
              </Text>
            </View>
          }
          renderItem={({item, index}) => {
            return isLoadingSellerState ? (
              <View key={index}>
                <NotificationItemSkeleton />
              </View>
            ) : (
              <Fragment key={item.id}>
                <Gap height={16} />
                <BaseNotif
                  image={item?.Product?.image_url}
                  status={item?.status}
                  title={item?.Product?.name}
                  price={item?.Product?.base_price}
                  bid={`${item?.price}`}
                  tanggal={item?.transaction_date}
                  onPress={() => navigation.navigate('InfoPenawar', {id: item.id})}
                  isRead={true}
                />
                <Gap height={16} />
                <View style={styles.divider} />
              </Fragment>
            );
          }}
        />
      );
    } else if (tabSelected === 'terjual') {
      return (
        <FlatList
          key={numColumns.terjual}
          numColumns={numColumns.terjual}
          style={styles.listDiminati}
          data={listData}
          showsVerticalScrollIndicator={false}
          keyExtractor={item => item.id}
          ListEmptyComponent={
            <View style={styles.emptyContent}>
              <Gap height={moderateScale(50)} />
              <ILPeople />
              <Gap height={moderateScale(16)} />
              <Text style={styles.emptyText}>
                Belum ada produkmu yang terjual nih, sabar ya rejeki nggak kemana kok
              </Text>
            </View>
          }
          renderItem={({item, index}) => {
            return isLoadingSellerState ? (
              <View key={index}>
                <NotificationItemSkeleton />
              </View>
            ) : (
              <Fragment key={item.id}>
                <Gap height={16} />
                <BaseNotif
                  image={item?.Product?.image_url}
                  status={item?.status}
                  title={item?.Product?.name}
                  price={item?.Product?.base_price}
                  bid={`${item?.price}`}
                  tanggal={item?.transact}
                  onPress={() => navigation.navigate('InfoPenawar', {id: item.id})}
                  isRead={true}
                />
                <Gap height={16} />
                <View style={styles.divider} />
              </Fragment>
            );
          }}
        />
      );
    }
  };

  return (
    <SafeAreaView style={styles.screen}>
      <View style={styles.heading}>
        <Text style={styles.header}>Daftar Jual Saya</Text>
        <Gap height={16} />
        <CardUser
          avatar={usersState?.profile?.image_url}
          name={usersState.profile?.full_name}
          city={usersState.profile?.city}
          button={true}
          onPressEdit={() => navigation.navigate('Profile')}
        />
        <Gap height={24} />
      </View>
      <View style={styles.category}>
        <ScrollView horizontal showsHorizontalScrollIndicator={false}>
          <CategoryButtonItem
            isActive={categorySelected === 'produk'}
            icon={categorySelected === 'produk' ? <ICBoxActive /> : <ICBox />}
            title="Produk"
            onPress={() => setCategorySelected('produk')}
          />
          <Gap width={moderateScale(16)} />
          <CategoryButtonItem
            isActive={categorySelected === 'diminati'}
            icon={categorySelected === 'diminati' ? <ICLoveActive /> : <ICLove />}
            title="Diminati"
            onPress={() => setCategorySelected('diminati')}
          />
          <Gap width={moderateScale(16)} />
          <CategoryButtonItem
            isActive={categorySelected === 'terjual'}
            icon={categorySelected === 'terjual' ? <ICDollarSignActive /> : <ICDollarSign />}
            title="Terjual"
            onPress={() => setCategorySelected('terjual')}
          />
        </ScrollView>
      </View>
      <Gap height={24} />
      {renderList(categorySelected)}
    </SafeAreaView>
  );
};

export default DaftarJual;

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: Colors.BACKGROUND,
  },
  header: {
    fontFamily: Fonts.PRIMARY.BOLD,
    fontSize: moderateScale(20),
    color: Colors.TEXT,
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  heading: {
    paddingHorizontal: moderateScale(20),
    paddingTop: moderateScale(20),
    paddingBottom: moderateScale(16),
  },
  category: {
    paddingHorizontal: moderateScale(20),
  },
  divider: {
    borderBottomColor: Colors.DISABLE,
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  listDiminati: {
    paddingHorizontal: moderateScale(20),
  },
  emptyContent: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  emptyText: {
    fontFamily: Fonts.PRIMARY.REGULAR,
    fontSize: moderateScale(14),
    color: Colors.SECONDARY,
    textAlign: 'center',
  },
});
