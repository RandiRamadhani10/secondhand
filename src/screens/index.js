import Login from './Login';
import Profile from './Profile';
import Daftar from './Daftar';
import TambahBarang from './TambahBarang';
import Akun from './Akun';
import DaftarJual from './DaftarJual';
import Home from './Home';
import Jual from './Jual';
import Notifikasi from './Notifikasi';
import Splash from './Splash';
import Detail from './Detail';
import DetailProduk from './DetailProduk';
import InfoPenawar from './InfoPenawar';

export {
  Login,
  Profile,
  Daftar,
  Akun,
  DaftarJual,
  Home,
  Jual,
  Notifikasi,
  Splash,
  TambahBarang,
  Detail,
  DetailProduk,
  InfoPenawar,
};
