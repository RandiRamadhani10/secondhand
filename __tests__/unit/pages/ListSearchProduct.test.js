import React from 'react';
import {create} from 'react-test-renderer';
import {createStore} from 'redux';

import {Home} from '../../../src/screens';
import ContainerTesting from '../../../src/utils/containerTesting/reduxTesting';
import {productsData} from '../../../src/utils/ProductData';

describe('Unit: List Search Product', () => {
  it('should render list search product with length 2', () => {
    const productState = (state = productsData, action) => {
      switch (action.type) {
        default:
          return state;
      }
    };

    const store = createStore(productState);

    const {root, toJSON} = create(ContainerTesting(<Home />, store));

    const compJSON = toJSON();

    const state = root.props.store.getState();
    expect(root.props.store).toBeTruthy();
    expect(state).toHaveLength(6);
    expect(compJSON).toMatchSnapshot();
  });

  it('should render empty list search product with length 0', () => {
    const productState = (state = [], action) => {
      switch (action.type) {
        default:
          return state;
      }
    };

    const store = createStore(productState);

    const {root, toJSON} = create(ContainerTesting(<Home />, store));

    const compJSON = toJSON();

    const state = root.props.store.getState();
    expect(root.props.store).toBeTruthy();
    expect(state).toHaveLength(0);
    expect(compJSON).toMatchSnapshot();
  });
});
