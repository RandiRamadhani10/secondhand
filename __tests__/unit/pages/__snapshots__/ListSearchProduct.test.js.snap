// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit: List Search Product should render empty list search product with length 0 1`] = `null`;

exports[`Unit: List Search Product should render list search product with length 2 1`] = `null`;
